### PR TITLE
add_routes() method to attach several HTTP methods to one handler

### PR DESCRIPTION
**Pull request by GMLudo**
Hi,

For `POST` and `PUT` methods, most of the time, we share business logic in our `aiohttp.web` daemons: We use `id` parameter in payload to know if it's a new object (empty id) or we need to update the database.

For now, we duplicate lines for POST and PUT, but for example, the biggest daemon we made has 100+ endpoints, it will reduce the number of lines we need to declare endpoints.
It's another method than `add_route`, to avoid that this feature could shoot your foot in your existing source code.
For documentation, maybe I should add another method documentation, as you like.

As I've discussed with @asvetlov during aiohttp PyCON sprint code, it's the latest feature we're missing compare to the fork of aiorest we've made. To remember, instead of to continue our fork, we've decided to abandon our fork, use `aiohttp.web` and promote this solution, to mutualize efforts.
To follow this idea, we wish to integrate as most as possible our HTTP practices with `aiohttp.web`.

Thanks for your feedbacks.

### CODE REVIEW

File: aiohttp/web_urldispatcher.py [hunk 1]
```diff
@@ -315,6 +315,12 @@ def register_route(self, route):
                 self._routes[name] = route
         self._urls.append(route)
 
+    def add_routes(self, methods, path, handler,
+                   *args, name=None, expect_handler=None):
```
Comment (asvetlov) [inline aiohttp/web_urldispatcher.py:319]: The method signature is ambiguous: when you call it with `name` parameter it should raise `ValueError` by https://github.com/KeepSafe/aiohttp/blob/master/aiohttp/web_urldispatcher.py#L311

Also the method should return list of `Route` instances retrieved by `.add_route()` call.

**Comment by fafhrd91** [comment]
is this required for aiohttp core? in master i added "*" method, with it you can make your own routing.

https://github.com/KeepSafe/aiohttp/blob/master/aiohttp/hdrs.py#L4

**Comment by asvetlov** [comment]
@fafhrd91 sorry, I've missed `*` method.
I think we can close the issue but please wait a bit for @GMLudo response.
Also `./docs/web.rst` should be updated as well as `./docs/web_reference` to mention a `'*'` for _method_ parameter first.

**Comment by ludovic-gasc** [comment]
@fafhrd91: f_kin' clever, it's exactly what we need: `*` HTTP method is interesting to have a catch all, especially when you've made a man-in-the-middle proxy daemon to rewrite step-by-step a legacy application.

Finally, with `*` HTTP method, only the use case CREATE/UPDATE (POST/PUT) is missing: it's clearly less important, we can live to double `add_route()` method calls, not necessary to risk to transform `aiohttp.web` as a Web framework bloatware, like you have with most Web frameworks ;-)

Thanks guys for your answers.

**Comment by asvetlov** [comment]
Documentation updated.

**Pull request closed without merging**